src/BrainCloudVirtualCurrency.cpp

```diff
@@ -16,29 +16,35 @@ namespace BrainCloud
 {
     BrainCloudVirtualCurrency::BrainCloudVirtualCurrency(BrainCloudClient* in_client) : m_client(in_client) { }
 
-    void BrainCloudVirtualCurrency::getCurrency(const std::string& in_vcId, IServerCallback * in_callback)
+    void BrainCloudVirtualCurrency::getCurrency(const char* in_vcId, IServerCallback * in_callback)
     {
-        Json::Value message;
-        message[OperationParam::VirtualCurrencyServiceVCId.getValue()] = in_vcId;
+        Json::Value message = Json::nullValue;
+        if (StringUtil::IsOptionalParameterValid(in_vcId)) {
+            message[OperationParam::VirtualCurrencyServiceVCId.getValue()] = in_vcId;
+        }
 
         ServerCall * sc = new ServerCall(ServiceName::VirtualCurrency, ServiceOperation::VirtualCurrencyGetCurrency, message, in_callback);
         m_client->sendRequest(sc);
     }
 
-    void BrainCloudVirtualCurrency::getParentCurrency(const std::string& in_vcId, const std::string& in_levelName, IServerCallback * in_callback)
+    void BrainCloudVirtualCurrency::getParentCurrency(const char* in_vcId, const std::string& in_levelName, IServerCallback * in_callback)
     {
         Json::Value message;
-        message[OperationParam::VirtualCurrencyServiceVCId.getValue()] = in_vcId;
+        if (StringUtil::IsOptionalParameterValid(in_vcId)) {
+            message[OperationParam::VirtualCurrencyServiceVCId.getValue()] = in_vcId;
+        }
         message[OperationParam::VirtualCurrencyServiceLevelName.getValue()] = in_levelName;
 
         ServerCall * sc = new ServerCall(ServiceName::VirtualCurrency, ServiceOperation::VirtualCurrencyGetParentCurrency, message, in_callback);
         m_client->sendRequest(sc);
     }
 
-    void BrainCloudVirtualCurrency::getPeerCurrency(const std::string& in_vcId, const std::string& in_peerCode, IServerCallback * in_callback)
+    void BrainCloudVirtualCurrency::getPeerCurrency(const char* in_vcId, const std::string& in_peerCode, IServerCallback * in_callback)
     {
         Json::Value message;
-        message[OperationParam::VirtualCurrencyServiceVCId.getValue()] = in_vcId;
+        if (StringUtil::IsOptionalParameterValid(in_vcId)) {
```

Comment:
You know this is a bit silly, but we can just test it like that:
```
if (in_vcId)
```
_Dont change anything, just pointing it out_